File: bench/Reference/binary_write_read.js
// Node.js reference: binary file read + write + read-back
const fs = require('fs');

const data = fs.readFileSync('bench/data/bench_binary.bin');
fs.writeFileSync('/tmp/js_bench_binary_wr.bin', data);
const data2 = fs.readFileSync('/tmp/js_bench_binary_wr.bin');
console.log(data2.length);

File: bench/Reference/binary_write_read.hs
-- Haskell reference: binary file read + write + read-back
import qualified Data.ByteString as BS

main :: IO ()
main = do
    dat <- BS.readFile "bench/data/bench_binary.bin"
    BS.writeFile "/tmp/hs_bench_binary_wr.bin" dat
    dat2 <- BS.readFile "/tmp/hs_bench_binary_wr.bin"
    print (BS.length dat2)
